diff --git a/node_modules/react-native-maps/android/src/main/java/com/rnmaps/maps/MapMarker.java b/node_modules/react-native-maps/android/src/main/java/com/rnmaps/maps/MapMarker.java
index b755f9b..d1ded1a 100644
--- a/node_modules/react-native-maps/android/src/main/java/com/rnmaps/maps/MapMarker.java
+++ b/node_modules/react-native-maps/android/src/main/java/com/rnmaps/maps/MapMarker.java
@@ -119,6 +119,23 @@ public class MapMarker extends MapFeature {
         }
       };
 
+  @Override
+  protected void onLayout(boolean changed, int l, int t, int r, int b) {
+    super.onLayout(changed, l, t, r, b);
+
+    int newWidth = r - l;
+    int newHeight = b - t;
+    if (newWidth > 0 && newHeight > 0 && (newWidth != this.width || newHeight != this.height)) {
+      this.width = newWidth;
+      this.height = newHeight;
+
+      clearDrawableCache();
+      if (marker != null) {
+        update(true);
+      }
+    }
+  }
+
   public MapMarker(Context context, MapMarkerManager markerManager) {
     super(context);
     this.context = context;
